### PR TITLE
feat: draft pipeline for AWS OS

### PR DESCRIPTION
**Pull request by ghost**
## Checklist

- [ ] for CI changes:
  - [ ] structural/foundational changes signed off by [CI DRI](https://github.com/cmur2)
  - [ ] [ci.yml](https://github.com/camunda/camunda/blob/main/.github/workflows/ci.yml) modifications comply with ["Unified CI" requirements](https://github.com/camunda/camunda/wiki/CI-&-Automation#workflow-inclusion-criteria)

## Related issues

- https://github.com/camunda/team-data-engine-maintenance/issues/49

### CODE REVIEW

File: .github/workflows/zeebe-aws-os-dispatch-tests.yml
Comment (cmur2) [inline .github/workflows/zeebe-aws-os-dispatch-tests.yml:1]: I'd use a different name since the tests this workflow is running are not all related to Zeebe. Maybe just "AWS OpenSearch Dispatch Tests" orso.

File: .github/workflows/zeebe-aws-os-dispatch-tests.yml
Comment (cmur2) [inline .github/workflows/zeebe-aws-os-dispatch-tests.yml:5]: This means the workflow is triggered by either humans or another workflow (which is not visible in this PR). It would be useful to know from where you want to trigger it, for a full review.

File: .github/workflows/zeebe-aws-os-dispatch-tests.yml
Comment (cmur2) [inline .github/workflows/zeebe-aws-os-dispatch-tests.yml:14]: This variable is unused as far as I can see, please remove when unused.

File: .github/workflows/zeebe-aws-os-dispatch-tests.yml
Comment (cmur2) [inline .github/workflows/zeebe-aws-os-dispatch-tests.yml:37]: Using `runs-on: ubuntu-latest` for all jobs means that *none of those tests* are running against a real AWS OpenSearch instance - the real AWS OpenSearch instance (available under `CAMUNDA_CI_OPENSEARCH_2_9_URL`) is only reachable from Camunda-hosted runners but not from GitHub.

I'm not sure if it is intentional to not use a real AWS OpenSearch instance?

File: .github/workflows/zeebe-aws-os-dispatch-tests.yml
Comment (cmur2) [inline .github/workflows/zeebe-aws-os-dispatch-tests.yml:82]: Apparently this setting doesn't take effect otherwise the job cannot be successful on Github-hosted runners (which don't have access to the real AWS OpenSearch instance provided under `CAMUNDA_CI_OPENSEARCH_2_9_URL`) - but also thir PR doesn't show if the job is successful, because it is not triggered.

File: .github/workflows/zeebe-aws-os-dispatch-tests.yml
Comment (cmur2) [inline .github/workflows/zeebe-aws-os-dispatch-tests.yml:5]: What is your goal, on which branches should this workflow run automatically? All branches? Just the `main` branch?

If the answer is all branches, then we should use [these triggers](https://github.com/camunda/camunda/blob/main/.github/workflows/ci.yml#L5-L10).
If the answer is just the `main` branch, then limit the existing `on.push` trigger accordingly.

File: .github/workflows/zeebe-aws-os-dispatch-tests.yml
Comment (cmur2) [inline .github/workflows/zeebe-aws-os-dispatch-tests.yml:69]: I'm not sure if this step of building the Docker image is actually needed. We cannot verify it now since the workflow is failing anyways, but let's keep this comment open for later (when the workflow is 🟢): then try the experiment if removing this step still makes this workflow green.

File: .github/workflows/zeebe-aws-os-dispatch-tests.yml
Comment (cmur2) [reply .github/workflows/zeebe-aws-os-dispatch-tests.yml:79]: The new CI job is failing right now. The error message is `Connect to localhost:443 [localhost/127.0.0.1, localhost/0:0:0:0:0:0:0:1] failed: Connection refused` which to me looks like the test cases assume that the AWS OpenSearch instance is available under `localhost` (which it is not).

I am still wondering if the `camunda.it.opensearch.aws_instance_url` parameter takes any effect. I tried to see if there is any code in the repository that uses/reads this parameter via [GitHub search but there is none](https://github.com/search?q=repo%3Acamunda%2Fcamunda+%22aws_instance_url%22&type=code).

@yevhen-cap where/how did you find that this parameter should be used? Any documentation or another engineer mentioning that it works? I don't see any evidence in the code that this is the right parameter.

File: .github/workflows/zeebe-aws-os-dispatch-tests.yml
Comment (yevhen-cap) [reply .github/workflows/zeebe-aws-os-dispatch-tests.yml:5]: At this moment on.push added for pipeline testing. will be removed before merge and on.workflow_dispatch would be only available option

File: .github/workflows/zeebe-aws-os-dispatch-tests.yml
Comment (yevhen-cap) [reply .github/workflows/zeebe-aws-os-dispatch-tests.yml:79]: Merged tests from branch, where this parameter present

File: .github/workflows/zeebe-aws-os-dispatch-tests.yml
Comment (igpetrov) [reply .github/workflows/zeebe-aws-os-dispatch-tests.yml:79]: For debug trace

**Comment by igpetrov** [comment]
Code changes moved from https://github.com/camunda/camunda/pull/25361. The reason: we wanted to make sure both PR and code fit each other, so in this specific case, we decided to union PRs

File: .github/workflows/zeebe-aws-os-dispatch-tests.yml
Comment (yevhen-cap) [reply .github/workflows/zeebe-aws-os-dispatch-tests.yml]: This step were removed

File: .github/workflows/zeebe-aws-os-dispatch-tests.yml
Comment (cmur2) [inline .github/workflows/zeebe-aws-os-dispatch-tests.yml:27]: (Not a blocker) A matrix build with exactly one group may be refactored to remove the unnecessary matrix setup, unless you plan future additions to the matrix.

File: .github/workflows/zeebe-aws-os-dispatch-tests.yml
Comment (cmur2) [inline .github/workflows/zeebe-aws-os-dispatch-tests.yml:34]: This setup (providing a locally available temporary Docker registry) is probably redundant now as well since the Docker image build step got removed.

File: .github/workflows/zeebe-aws-os-dispatch-tests.yml
Comment (igpetrov) [reply .github/workflows/zeebe-aws-os-dispatch-tests.yml:27]: Indeed, planned at least 3 more modules, but likely 5

**Review by igpetrov — APPROVED** [review]
LGTM, thank you

---

Let's wait for Camunda engineers approve before merge.

**Review by cmur2 — APPROVED** [review]
Assuming the current [test run](https://github.com/camunda/camunda/actions/runs/12182045073/job/33980102422?pr=24986) passes, the CI code looks also good to me!

**Comment by igpetrov** [comment]
Merging for now to unblock test development of another modules. If there are issues, happy to address them in another PR.

cc: @ChrisKujawa

**Review by ChrisKujawa — COMMENTED** [review]
Better late then never I guess :) LGTM. Just two comments (for potential improvement)

File: zeebe/exporters/opensearch-exporter/src/test/java/io/camunda/zeebe/exporter/opensearch/SearchDBExtension.java
Comment (igpetrov) [reply zeebe/exporters/opensearch-exporter/src/test/java/io/camunda/zeebe/exporter/opensearch/SearchDBExtension.java:20]: Addressed here: https://github.com/camunda/camunda/pull/25802

File: zeebe/exporters/opensearch-exporter/src/test/java/io/camunda/zeebe/exporter/opensearch/OpensearchClientIT.java
Comment (igpetrov) [reply zeebe/exporters/opensearch-exporter/src/test/java/io/camunda/zeebe/exporter/opensearch/OpensearchClientIT.java:118]: Addressed here: https://github.com/camunda/camunda/pull/25802

**Pull request merged**
